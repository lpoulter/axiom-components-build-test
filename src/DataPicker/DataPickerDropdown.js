import PropTypes from 'prop-types';
import { Component } from 'react';

export const DataPickerDropdownRef = 'DataPickerDropdown';

export default class DataPickerDropdown extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  static typeRef = DataPickerDropdownRef;

  render() {
    return this.props.children || null;
  }
}
