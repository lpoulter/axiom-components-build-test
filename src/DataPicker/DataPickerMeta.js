import PropTypes from 'prop-types';
import { Component } from 'react';

export const DataPickerMetaRef = 'DataPickerMeta';

export default class DataPickerMeta extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  static typeRef = DataPickerMetaRef;

  render() {
    return this.props.children || null;
  }
}

