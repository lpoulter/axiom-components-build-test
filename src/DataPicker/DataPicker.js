import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { findComponent } from '@brandwatch/axiom-utils';
import atIds from '@brandwatch/axiom-automation-testing/ids';
import Card from '../Card/Card';
import CardContent from '../Card/CardContent';
import ColorPicker from '../ColorPicker/ColorPicker';
import ColorPickerOption from '../ColorPicker/ColorPickerOption';
import Dropdown from '../Dropdown/Dropdown';
import DropdownContext from '../Dropdown/DropdownContext';
import DropdownSource from '../Dropdown/DropdownSource';
import DropdownTarget from '../Dropdown/DropdownTarget';
import Grid from '../Grid/Grid';
import GridCell from '../Grid/GridCell';
import Heading from '../Typography/Heading';
import Icon from '../Icon/Icon';
import Link from '../Typography/Link';
import { DataPickerDropdownRef } from './DataPickerDropdown';
import { DataPickerMetaRef } from './DataPickerMeta';
import './DataPicker.css';

export default class DataPicker extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    /** Selected color that will be displayed in the header */
    color: PropTypes.oneOf([
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ]),
    /** List of pickable colors to override the default, which is all colors. */
    colorOptions: PropTypes.arrayOf(PropTypes.string),
    /** List of disabled colors that become unpickable */
    disabledColors: PropTypes.arrayOf(PropTypes.string),
    /** Handler that is called when the color picker is opened */
    onColorPickerOpen: PropTypes.func,
    /** Handler that is called when a color is selected */
    onSelectColor: PropTypes.func,
    /** Placeholder text that will be displayed when no value is provided */
    placeholder: PropTypes.string.isRequired,
    /** Value that will be displayed in the header. */
    value: PropTypes.string,
  };

  render() {
    const {
      children,
      color,
      colorOptions,
      disabledColors,
      onColorPickerOpen,
      onSelectColor,
      placeholder,
      value,
      ...rest
    } = this.props;

    const dropdownMenu = findComponent(children, DataPickerDropdownRef);
    const metaData = findComponent(children, DataPickerMetaRef);
    const title = value || placeholder;

    return (
      <Card { ...rest }
          border={ Boolean(dropdownMenu) }
          shade={ dropdownMenu ? 'shade-1' : 'shade-3' }
          size="medium">
        <div className="ax-data-picker">
          <div className="ax-data-picker__dropdown">
            <CardContent size="small">
              <Grid
                  gutters="tiny"
                  responsive={ false }
                  verticalAlign="middle"
                  wrap={ false }>
                <GridCell none>
                  { onSelectColor ? (
                    <ColorPicker
                        colorOptions={ colorOptions }
                        disabledOptions={ disabledColors }
                        onOpen={ onColorPickerOpen }
                        onSelectColor={ onSelectColor }
                        selected={ color } />
                  ) : (
                    <ColorPickerOption color={ color } />
                  ) }
                </GridCell>

                <GridCell>
                  { dropdownMenu && (
                    <Dropdown>
                      <DropdownTarget>
                        <div
                            className="ax-data-picker__link"
                            data-ax-at={ atIds.DataPicker.dropdown }>
                          <Link style="body">
                            <Grid responsive={ false } verticalAlign="middle" wrap={ false }>
                              <GridCell>
                                <Heading textEllipsis title={ title }>
                                  { title }
                                </Heading>
                              </GridCell>

                              <GridCell none>
                                <Icon name="chevron-down"/>
                              </GridCell>
                            </Grid>
                          </Link>
                        </div>
                      </DropdownTarget>

                      <DropdownSource>
                        <DropdownContext>
                          { dropdownMenu }
                        </DropdownContext>
                      </DropdownSource>
                    </Dropdown>
                  ) }

                  { !dropdownMenu && (
                    <Heading textEllipsis title={ title }>
                      { title }
                    </Heading>
                  ) }
                </GridCell>
              </Grid>
            </CardContent>
          </div>

          { metaData && (
            <div className="ax-data-picker__meta">
              <CardContent size="small">
                { metaData }
              </CardContent>
            </div>
          ) }
        </div>
      </Card>
    );
  }
}

