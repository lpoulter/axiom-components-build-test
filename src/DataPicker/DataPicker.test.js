import React from 'react';
import renderer from 'react-test-renderer';
import DataPicker from './DataPicker';
import DataPickerDropdown from './DataPickerDropdown';
import DataPickerMeta from './DataPickerMeta';

const getComponent = (props = {}) =>
  renderer.create(
    <DataPicker { ...props } placeholder="Lorem ipsum">
      { props.children || [
        <DataPickerDropdown key="dropdown" />,
        <DataPickerMeta key="meta" />,
      ] }
    </DataPicker>
  );

describe('DataPicker', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with just DataPickerDropdown', () => {
    const component = getComponent({
      children: <DataPickerDropdown />,
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with just DataPickerMeta', () => {
    const component = getComponent({
      children: <DataPickerMeta />,
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with onSelectColor', () => {
    const component = getComponent({ onSelectColor: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with value', () => {
    const component = getComponent({ value: 'Dolar amor' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

