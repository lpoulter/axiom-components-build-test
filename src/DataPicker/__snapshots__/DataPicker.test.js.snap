// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPicker renders with defaultProps 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--border ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-data-picker"
  >
    <div
      className="ax-data-picker__dropdown"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      >
        <div
          className="ax-grid__container ax-space--x6"
        >
          <div
            className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle"
          >
            <div
              className="ax-grid__cell ax-grid__cell--none"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-color-picker__option ax-color-picker__option--empty"
                onClick={[Function]}
                style={
                  Object {
                    "height": ".75rem",
                    "width": ".75rem",
                  }
                }
              />
            </div>
            <div
              className="ax-grid__cell"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-data-picker__link"
                data-ax-at="9c4d25dadaa7"
                onClick={[Function]}
                onFocus={[Function]}
              >
                <a
                  className="ax-link ax-link--style-body"
                >
                  <div
                    className="ax-grid__container ax-space--x6"
                  >
                    <div
                      className="ax-grid ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-middle"
                    >
                      <div
                        className="ax-grid__cell"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <h5
                          className="ax-space--x2 ax-text--ellipsis ax-text--size-large"
                          title="Lorem ipsum"
                        >
                          Lorem ipsum
                        </h5>
                      </div>
                      <div
                        className="ax-grid__cell ax-grid__cell--none"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <svg
                          className="ax-icon ax-icon--chevron-down"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "",
                            }
                          }
                          style={
                            Object {
                              "height": "1rem",
                              "width": "1rem",
                            }
                          }
                          viewBox="0 0 16 16"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ax-data-picker__meta"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      />
    </div>
  </div>
</div>
`;

exports[`DataPicker renders with just DataPickerDropdown 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--border ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-data-picker"
  >
    <div
      className="ax-data-picker__dropdown"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      >
        <div
          className="ax-grid__container ax-space--x6"
        >
          <div
            className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle"
          >
            <div
              className="ax-grid__cell ax-grid__cell--none"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-color-picker__option ax-color-picker__option--empty"
                onClick={[Function]}
                style={
                  Object {
                    "height": ".75rem",
                    "width": ".75rem",
                  }
                }
              />
            </div>
            <div
              className="ax-grid__cell"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-data-picker__link"
                data-ax-at="9c4d25dadaa7"
                onClick={[Function]}
                onFocus={[Function]}
              >
                <a
                  className="ax-link ax-link--style-body"
                >
                  <div
                    className="ax-grid__container ax-space--x6"
                  >
                    <div
                      className="ax-grid ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-middle"
                    >
                      <div
                        className="ax-grid__cell"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <h5
                          className="ax-space--x2 ax-text--ellipsis ax-text--size-large"
                          title="Lorem ipsum"
                        >
                          Lorem ipsum
                        </h5>
                      </div>
                      <div
                        className="ax-grid__cell ax-grid__cell--none"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <svg
                          className="ax-icon ax-icon--chevron-down"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "",
                            }
                          }
                          style={
                            Object {
                              "height": "1rem",
                              "width": "1rem",
                            }
                          }
                          viewBox="0 0 16 16"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DataPicker renders with just DataPickerMeta 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--shade-3 ax-card--border-radius-small"
>
  <div
    className="ax-data-picker"
  >
    <div
      className="ax-data-picker__dropdown"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      >
        <div
          className="ax-grid__container ax-space--x6"
        >
          <div
            className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle"
          >
            <div
              className="ax-grid__cell ax-grid__cell--none"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-color-picker__option ax-color-picker__option--empty"
                onClick={[Function]}
                style={
                  Object {
                    "height": ".75rem",
                    "width": ".75rem",
                  }
                }
              />
            </div>
            <div
              className="ax-grid__cell"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <h5
                className="ax-space--x2 ax-text--ellipsis ax-text--size-large"
                title="Lorem ipsum"
              >
                Lorem ipsum
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ax-data-picker__meta"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      />
    </div>
  </div>
</div>
`;

exports[`DataPicker renders with onSelectColor 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--border ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-data-picker"
  >
    <div
      className="ax-data-picker__dropdown"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      >
        <div
          className="ax-grid__container ax-space--x6"
        >
          <div
            className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle"
          >
            <div
              className="ax-grid__cell ax-grid__cell--none"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-color-picker__option ax-color-picker__option--empty ax-color-picker__option--clickable"
                data-ax-at="98df24c31c96"
                onClick={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "height": ".75rem",
                    "width": ".75rem",
                  }
                }
              />
            </div>
            <div
              className="ax-grid__cell"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-data-picker__link"
                data-ax-at="9c4d25dadaa7"
                onClick={[Function]}
                onFocus={[Function]}
              >
                <a
                  className="ax-link ax-link--style-body"
                >
                  <div
                    className="ax-grid__container ax-space--x6"
                  >
                    <div
                      className="ax-grid ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-middle"
                    >
                      <div
                        className="ax-grid__cell"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <h5
                          className="ax-space--x2 ax-text--ellipsis ax-text--size-large"
                          title="Lorem ipsum"
                        >
                          Lorem ipsum
                        </h5>
                      </div>
                      <div
                        className="ax-grid__cell ax-grid__cell--none"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <svg
                          className="ax-icon ax-icon--chevron-down"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "",
                            }
                          }
                          style={
                            Object {
                              "height": "1rem",
                              "width": "1rem",
                            }
                          }
                          viewBox="0 0 16 16"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ax-data-picker__meta"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      />
    </div>
  </div>
</div>
`;

exports[`DataPicker renders with value 1`] = `
<div
  className="ax-card ax-card--size-medium ax-card--border ax-card--shade-1 ax-card--border-radius-small"
>
  <div
    className="ax-data-picker"
  >
    <div
      className="ax-data-picker__dropdown"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      >
        <div
          className="ax-grid__container ax-space--x6"
        >
          <div
            className="ax-grid ax-grid--gutters-horizontal--tiny ax-grid--gutters-vertical--tiny ax-grid--vertical-middle"
          >
            <div
              className="ax-grid__cell ax-grid__cell--none"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-color-picker__option ax-color-picker__option--empty"
                onClick={[Function]}
                style={
                  Object {
                    "height": ".75rem",
                    "width": ".75rem",
                  }
                }
              />
            </div>
            <div
              className="ax-grid__cell"
              style={
                Object {
                  "width": undefined,
                }
              }
            >
              <div
                className="ax-data-picker__link"
                data-ax-at="9c4d25dadaa7"
                onClick={[Function]}
                onFocus={[Function]}
              >
                <a
                  className="ax-link ax-link--style-body"
                >
                  <div
                    className="ax-grid__container ax-space--x6"
                  >
                    <div
                      className="ax-grid ax-grid--gutters-horizontal--medium ax-grid--gutters-vertical--medium ax-grid--vertical-middle"
                    >
                      <div
                        className="ax-grid__cell"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <h5
                          className="ax-space--x2 ax-text--ellipsis ax-text--size-large"
                          title="Dolar amor"
                        >
                          Dolar amor
                        </h5>
                      </div>
                      <div
                        className="ax-grid__cell ax-grid__cell--none"
                        style={
                          Object {
                            "width": undefined,
                          }
                        }
                      >
                        <svg
                          className="ax-icon ax-icon--chevron-down"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "",
                            }
                          }
                          style={
                            Object {
                              "height": "1rem",
                              "width": "1rem",
                            }
                          }
                          viewBox="0 0 16 16"
                        />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ax-data-picker__meta"
    >
      <div
        className="ax-card__content ax-card__content--separator-solid ax-card__content--size-small"
      />
    </div>
  </div>
</div>
`;
